### PR TITLE
Generated trigonometric functions for Variable

### PR DESCRIPTION
**Pull request by jl-wynen**
I'm not terribly happy with this solution but it gets the job done and is fairly flexible.

This also fixes an error reported by MSVC that the forward declaration in `variable_unary.cpp.in` had the wrong visibility.

### CODE REVIEW

File: variable/trigonometry.cpp [hunk 1]
```diff
@@ -15,76 +12,4 @@ using namespace scipp::core;
...
-Variable atan2(const Variable &y, const Variable &x) {
-  return transform(y, x, element::atan2, "atan2");
-}
-
-Variable &atan2(const Variable &y, const Variable &x, Variable &out) {
-  transform_in_place(out, y, x, element::atan2_out_arg, "atan2");
-  return out;
-}
-
 } // namespace scipp::variable
```
Comment (SimonHeybrock) [inline variable/trigonometry.cpp:15]: Can remove this file?

File: templates/variable_binary.cpp.in [hunk 2]
```diff
@@ -6,12 +6,36 @@
...
 Variable @NAME@(const Variable &a, const Variable &b) {
-  return transform(a, b, element::@OPNAME@, std::string_view("@OPNAME@"));
+  return transform(preprocess(a), preprocess(b), element::@OPNAME@,
+                   std::string_view("@OPNAME@"));
+}
+
+#ifdef GENERATE_OUT
+Variable &@NAME@(const Variable &a, const Variable &b, Variable &out) {
+  transform_in_place(
+      out, preprocess(a), preprocess(b), assign_unary { element::@OPNAME@ },
```
Comment (SimonHeybrock) [inline templates/variable_binary.cpp.in:35]: `assign_unary` but the op is binary?

**Review by SimonHeybrock — COMMENTED** [review]
Looks generally good, comments:

File: templates/variable_binary.cpp.in (hunk 2 above)
Comment (jl-wynen) [reply templates/variable_binary.cpp.in:35]: I don't remember why we called it `assign_unary`. It seems to work with binary functions as well.

File: variable/trigonometry.cpp (hunk 1 above)
Comment (jl-wynen) [reply variable/trigonometry.cpp:15]: Right, forgot to do that.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**